rasterio/_base.pyx

```diff
@@ -306,6 +307,10 @@ cdef class DatasetBase:
             # Read-only + Rasters + Sharing + Errors
             flags = 0x00 | 0x02 | sharing_flag | 0x40
 
+            # Threadsafe support added in GDAL 3.10 for Read-only rasters
+            if get_gdal_version_info("VERSION_NUM") >= "310":
```

Comment:
automatically adding the thread_safe flag should be benchmarked. For all drivers (besides the LIBERTIFF one), there will be "some" overhead cost associated. It could be visible in scenarios where you would loop over many small datasets

Reply:
Thanks for the insight! I might just enable threadsafe if libertiff is in the driver list. I really only have geotiffs to test.